apps/backend/test/unit/reconciliation/pos-reconciliation.spec.ts

```diff
@@ -1,58 +1,311 @@
+    posDepositService = module.get(PosDepositService);
+    paymentService = module.get(PaymentService);
+    logger = module.get(Logger);
+  });
+
+  afterAll(() => {
+    jest.resetAllMocks();
   });
 
   it('should be defined', () => {
     expect(service).toBeDefined();
+    expect(posDepositService).toBeDefined();
+    expect(paymentService).toBeDefined();
+    expect(logger).toBeDefined();
+  });
+
+  it('should verify the test data transaction_time', () => {
+    expect(
+      differenceInMinutes(
+        parse(payments[0].transaction.transaction_time, 'HH:mm:ss', new Date()),
+        parse(deposits[0].transaction_time, 'HH:mm:ss', new Date())
+      )
+    ).toBeGreaterThanOrEqual(20);
+  });
+
+  it('should verify the test data status is PENDING', () => {
+    expect(payments.every((itm) => itm.status === MatchStatus.PENDING)).toBe(
+      true
+    );
+    expect(deposits.every((itm) => itm.status === MatchStatus.PENDING)).toBe(
+      true
+    );
+  });
+
+  describe('matchPaymentsToDeposits heurisitics round one', () => {
+    it('should match payments to deposits according to first round of heuristics', () => {
+      const spy = jest.spyOn(service, 'matchPosPaymentToPosDeposits');
+      const roundOneTimeDiff = 5;
+      const timeDiff = roundOneTimeDiff;
+
+      /**
+       * Call function to match based on first round of heuristics
+       */
+      matchedRoundOne = service.matchPosPaymentToPosDeposits(
+        payments,
+        deposits,
+        timeDiff
+      );
+      matches = [...matchedRoundOne];
+      expect(spy).toBeCalled();
+    });
+
+    it('should match time', () => {
+      matchedRoundOne.forEach((itm) => {
+        expect(
+          differenceInMinutes(
+            parse(
+              itm.payment.transaction.transaction_time,
+              'HH:mm:ss',
+              new Date()
+            ),
+            parse(itm.deposit.transaction_time, 'HH:mm:ss', new Date())
+          )
+        ).toBeLessThanOrEqual(5);
+      });
+    });
+  });
+
+  describe('should match payments to deposits according to second round of heuristics', () => {
+    it('should call matchPosPaymentToPosDeposits', () => {
+      const spy = jest.spyOn(service, 'matchPosPaymentToPosDeposits');
+      const roundTwoTimeDiff = 1440;
+      const timeDiff = roundTwoTimeDiff;
+      /**
+       * Verify that there are still pending payments and deposits
+       */
+      expect(payments.some((itm) => itm.status === MatchStatus.PENDING)).toBe(
+        true
+      );
+      expect(deposits.some((itm) => itm.status === MatchStatus.PENDING)).toBe(
+        true
+      );
+      /**
+       * Call function to match based on the second round of heuristics
+       */
+      matchedRoundTwo = service.matchPosPaymentToPosDeposits(
+        payments,
+        deposits,
+        timeDiff
+      );
+      matches.push(...matchedRoundTwo);
+      expect(spy).toBeCalled();
```

Comment:
I'm still wondering if there's a scenario in which this fails. And if so, how this is different from the other `matchedRoundOne` test, since it's the same function. I might need to read more on how the heuristics are working